src/transformers/pipelines/zero_shot_classification.py

```diff
@@ -65,24 +65,42 @@ class ZeroShotClassificationPipeline(Pipeline):
     def __init__(self, args_parser=ZeroShotClassificationArgumentHandler(), *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._args_parser = args_parser
-        if self.entailment_id == -1:
+        if self.model.config.model_type == "t5":
+            self._get_entailment_neutral_contradiction_token_id()
+        elif self.entailment_id == -1:
             logger.warning(
                 "Failed to determine 'entailment' label id from the label2id mapping in the model config. Setting to "
                 "-1. Define a descriptive label2id mapping in the model config to ensure correct outputs."
             )
 
     @property
     def entailment_id(self):
+        if self.model.config.model_type == "t5":
+            return -1
         for label, ind in self.model.config.label2id.items():
             if label.lower().startswith("entail"):
                 return ind
         return -1
 
+    def _get_entailment_neutral_contradiction_token_id(self):
+        class_ids = self.tokenizer(["entailment", "neutral", "contradiction"]).input_ids
+        self.entailment_token_id, self.neutral_token_id, self.contradiction_token_id = [ids[0] for ids in class_ids]
+        assert (
+            (self.entailment_token_id != self.neutral_token_id)
+            and (self.entailment_token_id != self.contradiction_token_id)
+            and (self.neutral_token_id != self.contradiction_token_id)
+        ), "The first token ids for 'Entailment', 'Neutral' and 'Contradiction' must be different."
```

Comment:
methods named `get_xxx` need to return something, you cannot assign to self, that's a side effect and VERY undesirable (especially with such a method name)

Reply:
Okay, this needs to be changed.